File: Assets/Flocking/Boid Controller.js
// these define the flock's behaviorvar minVelocity : float 	= 5;var maxVelocity : float 	= 20;
var randomness : float 		= 1; var flockSize : int			= 20;
var prefab : GameObject; 
var chasee : GameObject;

var flockCenter : Vector3;
var flockVelocity : Vector3;

private var boids;
function Start() {
	boids = new Array(flockSize);
	for (var i=0; i<flockSize; i++) {		var position = Vector3(						Random.value*GetComponent.<Collider>().bounds.size.x,						Random.value*GetComponent.<Collider>().bounds.size.y,						Random.value*GetComponent.<Collider>().bounds.size.z)-GetComponent.<Collider>().bounds.extents;		var boid = Instantiate(prefab, transform.position, transform.rotation);		boid.transform.parent = transform;		boid.transform.localPosition = position;
		boid.GetComponent("Boid Flocking").setController(gameObject);
		boids[i] = boid;
	}}function Update () {   
   	var theCenter = Vector3.zero;   	var theVelocity = Vector3.zero;
   	for (var boid : GameObject in boids) {
		theCenter       = theCenter + boid.transform.localPosition;		theVelocity     = theVelocity + boid.GetComponent.<Rigidbody>().velocity;
   	}
	flockCenter = theCenter/(flockSize);	
	flockVelocity = theVelocity/(flockSize);
}

File: Assets/Flocking/BoidController.cs
using UnityEngine;
using System.Collections;
using System.Collections.Generic;

/// <summary>
/// these define the flock's behavior
/// </summary>
public class BoidController : MonoBehaviour
{
	public float minVelocity = 5;
	public float maxVelocity = 20;
	public float randomness = 1;
	public int flockSize = 20;
	public BoidFlocking prefab;
	public Transform target;

	internal Vector3 flockCenter;
	internal Vector3 flockVelocity;

	List<BoidFlocking> boids = new List<BoidFlocking>();

	void Start()
	{
		for (int i = 0; i < flockSize; i++)
		{
			BoidFlocking boid = Instantiate(prefab, transform.position, transform.rotation) as BoidFlocking;
			boid.transform.parent = transform;
			boid.transform.localPosition = new Vector3(
							Random.value * GetComponent<Collider>().bounds.size.x,
							Random.value * GetComponent<Collider>().bounds.size.y,
							Random.value * GetComponent<Collider>().bounds.size.z) - GetComponent<Collider>().bounds.extents;
			boid.controller = this;
			boids.Add(boid);
		}
	}

	void Update()
	{
		Vector3 center = Vector3.zero;
		Vector3 velocity = Vector3.zero;
		foreach (BoidFlocking boid in boids)
		{
			center += boid.transform.localPosition;
			velocity += boid.GetComponent<Rigidbody>().velocity;
		}
		flockCenter = center / flockSize;
		flockVelocity = velocity / flockSize;
	}
}